Upstream-Status: Submitted

Picked from mesa-dev mailing list: http://marc.info/?l=mesa3d-dev&m=133053004130468

From: Benjamin Franzke <benjaminfranzke@googlemail.com>
Date: Wed, 29 Feb 2012 15:36:23 +0100
Subject: [PATCH] configure: Fix egl compilation without x11 headers

We dont want eglplatform.h to typedef egl native types
to x11 types, when x11 headers are not available.
---
 configure.ac |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index b384b44..8eb8459 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1262,6 +1262,10 @@ if test "x$enable_egl" = xyes; then
 
     AC_CHECK_FUNC(mincore, [DEFINES="$DEFINES -DHAVE_MINCORE"])
 
+    if test "$have_x" != yes; then
+        DEFINES="$DEFINES -DMESA_EGL_NO_X11_HEADERS"
+    fi
+
     if test "$enable_static" != yes; then
         # build egl_glx when libGL is built
         if test "x$enable_glx" = xyes; then
-- 
1.7.10.4

